vendor/bat-native-ads/src/bat/ads/internal/country_subdivision.cc

```diff
@@ -0,0 +1,195 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/country_subdivision.h"
+
+#include <stdint.h>
+#include <string>
+#include <sstream>
+
+#include "base/json/json_reader.h"
+#include "base/time/time.h"
+#include "brave/components/l10n/common/locale_util.h"
+#include "brave/components/l10n/browser/locale_helper.h"
+
+#include "bat/ads/internal/logging.h"
+#include "bat/ads/internal/static_values.h"
+#include "bat/ads/internal/time_util.h"
+
+using std::placeholders::_1;
+using std::placeholders::_2;
+using std::placeholders::_3;
+
+namespace ads {
+
+CountrySubdivision::CountrySubdivision(
+    AdsClient* ads_client)
+    : ads_client_(ads_client) {
+  BuildUrl();
+}
+
+CountrySubdivision::~CountrySubdivision() = default;
+
+void CountrySubdivision::FetchCountrySubdivisionIfAllowedForRegion() {
+  auto locale = brave_l10n::LocaleHelper::GetInstance()->GetLocale();
```

Comment:
nit: could we only use `auto` for complex types to make the code more readable. other `auto`'s may need to be fixed too

Reply:
changed: https://github.com/brave/brave-core/pull/5602/files#diff-c33be94bc446f2eda08f8ce0383f81b2R39